lib/commands.py

```diff
@@ -495,6 +497,70 @@ def paytomany(self, outputs, fee=None, from_addr=None, change_addr=None, nocheck
         tx = self._mktx(outputs, tx_fee, change_addr, domain, nocheck, unsigned, password, locktime)
         return tx.as_dict()
 
+    def _mktx_slp(self, token_id, destination, amount, fee, change_addr, domain, unsigned, password, locktime):
+        ''' This code is basically lifted from main_window.py 'do_update_fee'
+        and 'read_send_tab' and modified to fit here. '''
+        selected_slp_coins, slp_op_return_msg = SlpCoinChooser.select_coins(self.wallet, token_id, amount, self.config, domain=domain)
+        DUST = self.wallet.dust_threshold()  # 546 satoshis
+        if not slp_op_return_msg:
+            raise RuntimeError('Unable to find suitable SLP coin')
+        bch_outputs = [ slp_op_return_msg ]
+        bch_outputs.append((destination.kind, destination, DUST))  # hack: addr.kind == bitoin TYPE always in Address class
+        token_outputs = slp.SlpMessage.parseSlpOutputScript(bch_outputs[0][1]).op_return_fields['token_output']
+        coins = self.wallet.get_spendable_coins(domain, self.config)
+        if len(token_outputs) > 1 and len(bch_outputs) < len(token_outputs):
+            """ start of logic copied from wallet.py """
+            addrs = self.wallet.get_change_addresses()[-self.wallet.gap_limit_for_change:]
+            if not change_addr:
+                if self.wallet.use_change and addrs:
+                    # New change addresses are created only after a few
+                    # confirmations.  Select the unused addresses within the
+                    # gap limit; if none take one at random
+                    change_addrs = [addr for addr in addrs if
+                                    self.wallet.get_num_tx(addr) == 0]
+                    if not change_addrs:
+                        import random
+                        change_addrs = [random.choice(addrs)]
+                        change_addr = change_addrs[0]
+                    elif len(change_addrs) > 1:
+                        change_addr = change_addrs[1]
+                    else:
+                        change_addr = change_addrs[0]
+                else:
+                    change_addr = coins[0]['address']
+            bch_outputs.append((change_addr.kind, change_addr, DUST))
+
+        tx = self.wallet.make_unsigned_transaction(coins, bch_outputs, self.config, fee, mandatory_coins=selected_slp_coins)
+        self.wallet.check_sufficient_slp_balance(slp.SlpMessage.parseSlpOutputScript(slp_op_return_msg[1]), self.config)
+        if locktime != None:
+            tx.locktime = locktime
+        if not unsigned:
+            run_hook('sign_tx', self.wallet, tx)
+            self.wallet.sign_transaction(tx, password)
```

Comment:
using wallet.sign_transaction bypasses the extra SLP validity check that we should always pre-signing any transaction type to double-check we're not burning SLP tokens.  This was added to main_window.sign_tx as shown in link below.

Just need to add this similar code:

https://github.com/simpleledger/Electron-Cash-SLP/blob/520f3ac0123ba53937db48cba697de8b265dc081/gui/qt/main_window.py#L2502

Reply:
Ah thanks. I was worried here too that maybe I'm not doing all the checks.  Will "check" this out and add this check.

Reply:
Ok, I added the check. Let me know if anything else!